<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2020-05-27" />

<title>factor_flash</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.13/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">pathway_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/pathway_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">factor_flash</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2020-05-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-06-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>pathway_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200323code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200323)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200323code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200323)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/gene_BF_116.rdata
</td>
<td style="text-align:left;">
output/gene_BF_116.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/gene_BF_955.rdata
</td>
<td style="text-align:left;">
output/gene_BF_955.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/summary_assoc_2snp_flash_b.rdata
</td>
<td style="text-align:left;">
output/summary_assoc_2snp_flash_b.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/summary_assoc_955_flash_b.rdata
</td>
<td style="text-align:left;">
output/summary_assoc_955_flash_b.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/summary_assoc_116_flash_b.rdata
</td>
<td style="text-align:left;">
output/summary_assoc_116_flash_b.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/rs9557207_gene_snp.csv
</td>
<td style="text-align:left;">
output/rs9557207_gene_snp.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/rs11616269_gene_snp.csv
</td>
<td style="text-align:left;">
output/rs11616269_gene_snp.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/p_r_all_ebi2.rdata
</td>
<td style="text-align:left;">
output/p_r_all_ebi2.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/p_r_955_ebi2.rdata
</td>
<td style="text-align:left;">
output/p_r_955_ebi2.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/p_r_116_ebi2.rdata
</td>
<td style="text-align:left;">
output/p_r_116_ebi2.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/p_r_all_cisebi2.rdata
</td>
<td style="text-align:left;">
output/p_r_all_cisebi2.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/p_r_955_cisebi2.rdata
</td>
<td style="text-align:left;">
output/p_r_955_cisebi2.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/pathway_analysis/output/p_r_116_cisebi2.rdata
</td>
<td style="text-align:left;">
output/p_r_116_cisebi2.rdata
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229pathwayanalysistree9fe2a81f3730e415217b0088f55c02ff1bb43947targetblank9fe2a81a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/pathway_analysis/tree/9fe2a81f3730e415217b0088f55c02ff1bb43947" target="_blank">9fe2a81</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229pathwayanalysistree9fe2a81f3730e415217b0088f55c02ff1bb43947targetblank9fe2a81a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/pathway_analysis/tree/9fe2a81f3730e415217b0088f55c02ff1bb43947" target="_blank">9fe2a81</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Deleted:    analysis/INID.Rmd
    Deleted:    analysis/regression.Rmd
    Deleted:    output/LV101_ldpr.jpeg
    Deleted:    output/LV5.jpeg
    Deleted:    output/nzero_variants.rdata
    Deleted:    output/snp001_le.jpeg

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/factor_flash.Rmd</code>) and HTML (<code>docs/factor_flash.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/9fe2a81f3730e415217b0088f55c02ff1bb43947/analysis/factor_flash.Rmd" target="_blank">9fe2a81</a>
</td>
<td>
XSun
</td>
<td>
2020-06-04
</td>
<td>
wflow_publish(c(“output/pval_all_cisebi2.jpeg”, “output/pval_955_cisebi2.jpeg”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/pathway_analysis/e0aeb6a4d45c2375ca30205119fa9182a2b666ba/docs/factor_flash.html" target="_blank">e0aeb6a</a>
</td>
<td>
XSun
</td>
<td>
2020-06-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/02750bd52c1b471358f1ab73157536bc6d2187de/analysis/factor_flash.Rmd" target="_blank">02750bd</a>
</td>
<td>
XSun
</td>
<td>
2020-06-01
</td>
<td>
wflow_publish(c(“output/pval_allgene_ebi2.jpeg”, “output/pval_955_ebi2.jpeg”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/pathway_analysis/0c4ea8f4f28e0b0f5d8224e3deb97c08d24e315b/docs/factor_flash.html" target="_blank">0c4ea8f</a>
</td>
<td>
XSun
</td>
<td>
2020-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/f22fc94ec8390f73f01539518a1f3fda85c83d81/analysis/factor_flash.Rmd" target="_blank">f22fc94</a>
</td>
<td>
XSun
</td>
<td>
2020-05-29
</td>
<td>
wflow_publish(c(“analysis/factor_flash.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/pathway_analysis/a82ba03b8ac6bed718b6ed98187ea09747a1a3e8/docs/factor_flash.html" target="_blank">a82ba03</a>
</td>
<td>
XSun
</td>
<td>
2020-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/3be86b1dfef728c972117d8c1ab7f175e522e4df/analysis/factor_flash.Rmd" target="_blank">3be86b1</a>
</td>
<td>
XSun
</td>
<td>
2020-05-29
</td>
<td>
wflow_publish(c(“analysis/factor_flash.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/pathway_analysis/dda0be7fcd86a2b0fd04afc5015f89cd1f2b56f5/docs/factor_flash.html" target="_blank">dda0be7</a>
</td>
<td>
XSun
</td>
<td>
2020-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/c5d40864e618ceb17e470c9c12ba08d7d5deaba9/analysis/factor_flash.Rmd" target="_blank">c5d4086</a>
</td>
<td>
XSun
</td>
<td>
2020-05-29
</td>
<td>
wflow_publish(c(“analysis/factor_flash.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/pathway_analysis/1f5355f947841dc7f9977dc8c8914c7329c9f4de/docs/factor_flash.html" target="_blank">1f5355f</a>
</td>
<td>
XSun
</td>
<td>
2020-05-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/59ed837be20900db2180076dad9d338b97afa5fc/analysis/factor_flash.Rmd" target="_blank">59ed837</a>
</td>
<td>
XSun
</td>
<td>
2020-05-27
</td>
<td>
wflow_publish(c(“analysis/factor_flash.Rmd”, “analysis/index.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/pathway_analysis/8af1b4e17e2bb1f9615986bd8851f3591195e80f/docs/factor_flash.html" target="_blank">8af1b4e</a>
</td>
<td>
XSun
</td>
<td>
2020-05-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/186fd5d0c826da40ff3ffcd2d29d22da3761f5d3/analysis/factor_flash.Rmd" target="_blank">186fd5d</a>
</td>
<td>
XSun
</td>
<td>
2020-05-27
</td>
<td>
wflow_publish(c(“analysis/factor_flash.Rmd”, “output/rs11616269_pval_flash_b.jpeg”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this part, we used <a href="https://github.com/stephenslab/flashr">‘flashr’</a>, methods for matrix factorization based on <a href="https://arxiv.org/abs/1802.06931">Empirical Bayes Matrix Factorization</a>, to extract factors from IDIN gene expression matrix and part of the IDIN gene expression matrix.We also performed factor analysis find if any factor associates with the two SNPs ( <a href="https://www.ncbi.nlm.nih.gov/snp/rs9557207">rs9557207</a> and <a href="https://www.ncbi.nlm.nih.gov/snp/rs11616269">rs11616269</a>) and EBI2 expression data.</p>
</div>
<div id="material-and-methods" class="section level2">
<h2>Material and Methods</h2>
<ol style="list-style-type: decimal">
<li>Data
<ul>
<li>All genes in IDIN(same with the genes in <a href="https://xsun1229.github.io/pathway_analysis/IDIN.html">IDIN-PLIER</a>)</li>
<li>Part of the IDIN genes (genes associated with rs9557207 and rs11616269, listed by <em>Bayesian detection of expression quantitative trait [Genetics, 2011]</em>, see the table below, evidence of associations :BF &gt; 10)</li>
<li>EBI2 expression data and genotype of variants are from GTEx data.</li>
</ul></li>
</ol>
<p><strong>rs11616269</strong> <div id="htmlwidget-25477d99f8699b15a5c8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-25477d99f8699b15a5c8">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13"],["OMA1","TRIM37","PARP10","WDR33","LOC285074","EPC1","WSB1","VNN2","PIWIL4","KIF23","XPOT","MOCS2","ZBTB34"],["ENSG00000162600","ENSG00000108395","ENSG00000178685","ENSG00000136709",null,"ENSG00000120616","ENSG00000109046","ENSG00000112303","ENSG00000134627","ENSG00000137807","ENSG00000184575","ENSG00000164172","ENSG00000177125"],[10.46,50.49,39.47,85.24,11.78,14.88,26.02,27.59,21.08,53.91,199.78,11.67,16.89]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>hgnc_symbol<\/th>\n      <th>ensembl_gene_id<\/th>\n      <th>BF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script></p>
<p><strong>rs9557207</strong> <div id="htmlwidget-510034f0db893e377f45" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-510034f0db893e377f45">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25"],["TAF12","MRPL22","GALNT10","HLA-DRA","IFIT1","SP140","SF3B14","OAS2","ANXA2","CKS1B","MOV10","LOC285074","CSTF3","C11orf67","PPP2R5C","AIG1","LOC643668","CHMP5","XPO7","SP140","THYN1","BTN3A3","ATF3","DHX9","PSMC1"],["ENSG00000120656","ENSG00000082515","ENSG00000164574","ENSG00000204287,ENSG00000230726,ENSG00000226260,ENSG00000234794,ENSG00000227993,ENSG00000228987,ENSG00000206308","ENSG00000185745","ENSG00000079263",null,"ENSG00000111335","ENSG00000182718","ENSG00000173207","ENSG00000155363",null,"ENSG00000176102",null,"ENSG00000078304","ENSG00000146416",null,"ENSG00000086065","ENSG00000130227","ENSG00000079263","ENSG00000151500","ENSG00000111801","ENSG00000162772","ENSG00000135829","ENSG00000100764"],[10.04,14.8,13.5,16.63,31.87,19.03,15.05,25,44.73,23.43,29.53,10.25,20.35,10.8,58.21,11.42,16.52,10,33.35,42.05,13.94,25.21,23.68,11.77,12.58]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>hgnc_symbol<\/th>\n      <th>ensembl_gene_id<\/th>\n      <th>BF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script></p>
<ol start="2" style="list-style-type: decimal">
<li>We used workflow showed in <a href="https://github.com/stephenslab/flashr">‘flashr’</a> to extract factors.</li>
</ol>
<pre class="r"><code>f = flash(Y)    ### Y is the gene expression matrix.
ldf = flash_get_ldf(f)</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Then, we used the backfitting algorithm to improve the fit (but it takes longer).</li>
</ol>
<pre class="r"><code>f.b = flash(Y, f_init = f, backfit=TRUE, greedy=FALSE)
ldf.b = flash_get_ldf(f.b)</code></pre>
<ol start="4" style="list-style-type: decimal">
<li><p>Perform association tests between the facotrs and two SNPs/EBI2 expression/cis-genetic component of EBI2 expression.</p></li>
<li><p>Besides, I performed association tests for the SNP-related gene expression and snp to find if they are associated with each other (single gene expression ~ single snp).</p></li>
</ol>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<p>We obtained 62 factors for all genes in IDIN using FLASH. For genes associated with rs9557207, we got 8 factors and for genes associated with rs11616269, we got 4.</p>
<div id="factors-snps." class="section level3">
<h3>Factors ~ SNPs.</h3>
<div id="all-genes-in-idin" class="section level4">
<h4>All genes in IDIN</h4>
<p>We performed linear regression for factors(extracted from all genes in IDIN) ~ snp.</p>
<pre class="r"><code>  ### k indicates the kth factors
  y &lt;- as.numeric(factors[,k])   ## kth factors
  x &lt;- as.numeric(geno_2snp[,1])   ## genotype of rs11616269
  
  pval_snp1[k] &lt;- summary(lm(y ~ x))$coefficients[2,4]  ## pval of F-statistics in y ~ x regression
  rsq_snp1[k] &lt;- summary(lm(y ~ x))$r.squared    ## r-squared of the regression</code></pre>
<p>We made a histogram to show the pvalue distribution.</p>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/rs11616269_pval_flash_b.jpeg" width="80%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/rs9557207_pval_flash_b.jpeg" width="80%">
</figure>
<p>The table below shows the detailed results of the linear regression.</p>
<p>Notations: <strong>pval</strong> is the pvalue of F-statistics in regression. <strong>rsq</strong> is the r-squared of the regression.</p>
<p>You can sort the table by clicking the arrow by the header of each column.</p>
<div id="htmlwidget-0dba913bd51e029ac868" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0dba913bd51e029ac868">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4","factor5","factor6","factor7","factor8","factor9","factor10","factor11","factor12","factor13","factor14","factor15","factor16","factor17","factor18","factor19","factor20","factor21","factor22","factor23","factor24","factor25","factor26","factor27","factor28","factor29","factor30","factor31","factor32","factor33","factor34","factor35","factor36","factor37","factor38","factor39","factor40","factor41","factor42","factor43","factor44","factor45","factor46","factor47","factor48","factor49","factor50","factor51","factor52","factor53","factor54","factor55","factor56","factor57","factor58","factor59","factor60","factor61","factor62"],[0.786350788121974,0.784476480078141,0.96743124766656,0.730852953302353,0.214072321130867,0.0281553972503257,0.506295421919775,0.185755779225637,0.862962162725637,0.189681947938819,0.172915772770843,0.694943800364378,0.892598483474719,0.129727688334202,0.905492963407706,0.777352534378885,0.101496872711271,0.345033887170312,0.0604739208453584,0.618483813995497,0.947836030711031,0.389240052889006,0.28272311131326,0.694189025654493,0.363416495672689,0.486073996122918,0.508660441257117,0.864505120125347,0.850918814975111,0.0235502968985271,0.29594387511551,0.952050475248384,0.630177461708596,0.194674660467428,0.60308542130239,0.50326157053404,0.652891099240431,0.324477314419361,0.265944566527824,0.872355822085768,0.41960752859566,0.538868128989994,0.311760237232429,0.589878652239165,0.452332796112419,0.651148500017498,0.636163826434951,0.695749477642727,0.13501109699873,0.463732303370705,0.979202301434153,0.727818177812576,0.406237868365143,0.99330677165937,0.0870730074494454,0.0657842064789793,0.508390523540678,0.843060902341864,0.422502533571605,0.0522694460633896,0.175521564798964,0.972976462142706],[0.000110060372915626,0.000112048895815196,2.49753450861195e-06,0.000177257495758346,0.00230994159745955,0.00719276899436202,0.000661527554754341,0.00261972613204867,4.46317704630325e-05,0.00257358967077398,0.0027788740600693,0.000230363548961697,2.73098413480125e-05,0.00343343154312209,2.11169326889243e-05,0.000119782048023214,0.0040097109683325,0.00133489459342285,0.00526599174123642,0.000371472930098819,6.41250656649663e-06,0.00110996698629533,0.00172727486324655,0.000231565425149075,0.00123676588093459,0.000726634781906281,0.000654200383566012,4.36226431199204e-05,5.29181041903799e-05,0.00765213994674689,0.00163518942826418,5.41700512217328e-06,0.000347235750476337,0.00251649953782795,0.000404962656038863,0.00067101322957492,0.000302987081368622,0.00145334496000709,0.00185228283738613,3.86716423343184e-05,0.000975425015349,0.000565594999838146,0.00153171489610418,0.000435161806867649,0.000845783761428821,0.000306252960151006,0.000335216279724038,0.000229084595720704,0.00334109820523599,0.00080397779218733,1.01811585227968e-06,0.000181443251881438,0.00103285011915285,1.05426534803255e-07,0.00437641737843213,0.00505861917686697,0.000655033663650009,5.87188344494026e-05,0.000963346344205562,0.00562773192698864,0.00274549731730196,1.71916924758178e-06],[0.690826452390967,0.669739936086867,0.836670796587723,0.000327850988059584,0.226200680173479,0.872964619396514,0.671973670576182,0.357348085771581,0.284372821492633,0.898256548142735,0.730684515183917,0.291243039110489,0.104648067462672,0.706440220199475,0.555072231216736,0.823680387271862,0.619517368206635,0.338131685028464,0.866696171663543,0.120037683061022,0.602278491700003,0.975257709509254,0.344778990259317,0.428613780696963,0.275865934347355,0.908075112834414,0.833242529565533,0.709413126188624,0.889467370526863,0.0367921641451217,0.623267640282057,0.696682802994502,0.363848202077769,0.798984657481526,0.273146412830919,0.156807199309963,0.359566059081742,0.852266762976636,0.841554357244203,0.164413416434993,0.0631274256891358,0.852002161262452,0.720973996470885,0.642231605720199,0.342276943831858,0.335516189884415,0.221435368092864,0.493325099394964,0.624596573486093,0.142543740250784,0.69635670838782,0.166263899569923,0.479285719317021,0.361089062427168,0.377728825004902,0.553791393696393,0.678230407124645,0.934383476767009,0.411997263656778,0.454200104467702,0.812171666428509,0.929109150658274],[0.000236963820241422,0.000272481858830118,6.36674348826062e-05,0.0191463943455476,0.00219174106704573,3.83004889696079e-05,0.00026858126122375,0.0012683950877949,0.00171548876781117,2.44929059717889e-05,0.000177488384224132,0.00166732141094167,0.00393715353238057,0.000212499662698788,0.000521674154766864,7.43732613464294e-05,0.000369289717783023,0.00137359029517837,4.22100408882872e-05,0.00361416915417885,0.000406768161379369,1.44107710502951e-06,0.00133630502033933,0.000938250973003437,0.00177721087770802,1.99737096890194e-05,6.64083997268926e-05,0.000208014024464013,2.89358157374402e-05,0.00651055440044855,0.000361435012614387,0.000227608138182318,0.00123454353238335,9.71524236479672e-05,0.00179744899914713,0.00299886297523258,0.00125674966440841,5.19544258576802e-05,5.98667695116356e-05,0.00289192501498933,0.00516005006219938,5.21428727223482e-05,0.000191084888355233,0.000323296577766621,0.001350224767972,0.00138853160718034,0.00223726023338674,0.00070277273244184,0.000358676772340933,0.0032162110279471,0.000228123372786677,0.00286675962336201,0.000749513684434232,0.00124881018172189,0.00116498418441037,0.00052505845994908,0.000257831355469553,1.01548506462737e-05,0.00100777588253113,0.000838822652844481,8.45893861847746e-05,1.18574544228516e-05]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pval_rs11616269<\/th>\n      <th>rsq_rs11616269<\/th>\n      <th>pval_rs9557207<\/th>\n      <th>rsq_rs9557207<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="genes-associated-with-rs9557207" class="section level4">
<h4>Genes associated with rs9557207</h4>
<p>We performed linear regression for factors ~ snp. We list the p-values and r-squared of each regression here.</p>
<div id="htmlwidget-95579c01c4852a1bf681" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-95579c01c4852a1bf681">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4","factor5","factor6","factor7","factor8"],[0.856521489304836,0.0994617405642982,0.0346250067217021,0.880852460901004,0.302918553444053,0.478657812399715,0.518074051626245,0.2033560314724],[4.90461291059094e-05,0.00406396530420383,0.006674653441343,3.37077637217978e-05,0.00159108246472958,0.000752784162650328,0.000626548704992091,0.00242482078501356]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pval_rs9557207<\/th>\n      <th>rsq_rs9557207<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="genes-associated-with-rs11616269" class="section level4">
<h4>Genes associated with rs11616269</h4>
<div id="htmlwidget-7337a1d03e4dc5979046" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7337a1d03e4dc5979046">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4"],[0.663491431288003,0.924604556090726,0.857516041207157,0.470464251376316],[0.000283995749103048,1.34370088504984e-05,4.83612269865758e-05,0.000781217872184231]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pval_rs11616269<\/th>\n      <th>rsq_rs11616269<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="gene-expression-snps" class="section level3">
<h3>Gene expression ~ SNPs</h3>
<p><em>It seems that the two snps are not associated with the gene expression(genes are reported in [Bayesian detection of expression quantitative trait (Genetics, 2011)]. So I’m considering the factor analysis results are reasonable.</em></p>
<ul>
<li>rs9557207</li>
</ul>
<div id="htmlwidget-683be954aee5f3159b5a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-683be954aee5f3159b5a">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30"],["TAF12","MRPL22","GALNT10","HLA-DRA","","","","","","","IFIT1","SP140","SF3B14","OAS2","ANXA2","CKS1B","MOV10","LOC285074","CSTF3","C11orf67","PPP2R5C","AIG1","LOC643668","CHMP5","XPO7","THYN1","BTN3A3","ATF3","DHX9","PSMC1"],["ENSG00000120656","ENSG00000082515","ENSG00000164574","ENSG00000204287","ENSG00000230726","ENSG00000226260","ENSG00000234794","ENSG00000227993","ENSG00000228987","ENSG00000206308","ENSG00000185745","ENSG00000079263",null,"ENSG00000111335","ENSG00000182718","ENSG00000173207","ENSG00000155363",null,"ENSG00000176102",null,"ENSG00000078304","ENSG00000146416",null,"ENSG00000086065","ENSG00000130227","ENSG00000151500","ENSG00000111801","ENSG00000162772","ENSG00000135829","ENSG00000100764"],[10.04,14.8,13.5,16.63,null,null,null,null,null,null,31.87,19.03,15.05,25,44.73,23.43,29.53,10.25,20.35,10.8,58.21,11.42,16.52,10,33.35,13.94,25.21,23.68,11.77,12.58],[0.26103695,0.15829375,0.28357012,0.72148123,null,null,null,null,null,null,0.48483359,0.92568002,null,0.66937122,0.48401605,0.19371782,0.96913374,null,0.28055338,null,0.93233371,0.03442368,null,0.07776859,0.41244737,0.41090319,0.55963766,0.50816073,0.6171315,0.54157894],[0.00189,0.00298,0.00172,0.000191,null,null,null,null,null,null,0.000732,1.31e-05,null,0.000274,0.000735,0.00253,2.25e-06,null,0.00175,null,1.08e-05,0.00669,null,0.00466,0.00101,0.00101,0.00051,0.000657,0.000375,0.000559]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>hgnc_symbol<\/th>\n      <th>ensembl_gene_id<\/th>\n      <th>BF<\/th>\n      <th>pval<\/th>\n      <th>rsq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>rs11616269</li>
</ul>
<div id="htmlwidget-4cd78be07865bf8d6c49" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4cd78be07865bf8d6c49">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13"],["OMA1","TRIM37","PARP10","WDR33","LOC285074","EPC1","WSB1","VNN2","PIWIL4","KIF23","XPOT","MOCS2","ZBTB34"],["ENSG00000162600","ENSG00000108395","ENSG00000178685","ENSG00000136709",null,"ENSG00000120616","ENSG00000109046","ENSG00000112303","ENSG00000134627","ENSG00000137807","ENSG00000184575","ENSG00000164172","ENSG00000177125"],[10.46,50.49,39.47,85.24,11.78,14.88,26.02,27.59,21.08,53.91,199.78,11.67,16.89],[0.5700097,0.7363221,0.8303526,0.469654,null,0.4974739,0.6170187,0.7406645,0.4499087,0.9941406,0.9425259,0.8015665,0.6115727],[0.000484,0.00017,6.89e-05,0.000784,null,0.00069,0.000375,0.000164,0.000856,8.09e-08,7.8e-06,9.48e-05,0.000387]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>hgnc_symbol<\/th>\n      <th>ensembl_gene_id<\/th>\n      <th>BF<\/th>\n      <th>pval<\/th>\n      <th>rsq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="factors-ebi2-expression" class="section level3">
<h3>Factors ~ EBI2 expression</h3>
<p>We made histograms to show the pvalue distribution and we also made a table to show the detailed information for each factor.</p>
<div id="factors-extracted-from-all-idin-genes." class="section level4">
<h4>62 factors extracted from all IDIN genes.</h4>
<p>We got 12 of 62 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)</p>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_allgene_ebi2.jpeg" width="80%">
</figure>
<div id="htmlwidget-1858cd90d56a4e65b6c0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1858cd90d56a4e65b6c0">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4","factor5","factor6","factor7","factor8","factor9","factor10","factor11","factor12","factor13","factor14","factor15","factor16","factor17","factor18","factor19","factor20","factor21","factor22","factor23","factor24","factor25","factor26","factor27","factor28","factor29","factor30","factor31","factor32","factor33","factor34","factor35","factor36","factor37","factor38","factor39","factor40","factor41","factor42","factor43","factor44","factor45","factor46","factor47","factor48","factor49","factor50","factor51","factor52","factor53","factor54","factor55","factor56","factor57","factor58","factor59","factor60","factor61","factor62"],[1.02198222191821e-72,1.02209707700027e-41,0.535347928820461,0.0820399926366768,0.0154200037196831,0.212237159743473,0.545031563650886,2.17448998177549e-13,0.00268817369307082,0.0576496375365858,0.00220204523640437,0.0366081205766917,0.10569928719761,0.0637728291351354,0.0814624158891095,0.583334482698317,0.577971826666585,0.544609093462732,0.0572549323873142,0.0218585675941027,0.095001470088022,0.0987827230387754,0.959319784822648,0.0502468226830674,0.743007670950199,0.209471482832672,0.466048247086732,0.838673369104356,0.143320757582563,0.75387991686274,0.663598159531635,0.485452368171887,0.89348537037819,0.298610360992955,0.249636472719819,0.325055470495618,0.966267152433803,0.643509769646538,0.259981076033001,0.331473225725873,0.795123336264051,0.779583249982413,0.923992458088808,0.612618320922409,0.269507495177948,0.00109780806127189,0.0225194649721066,0.0339877931133498,0.0104989226640211,0.915128142552099,0.95001735212856,0.897073995246926,0.665120099855902,0.651567871444056,0.855996514730031,0.687884845039665,0.590743847193656,0.80914724477432,0.919114340439834,0.257252015233522,0.866264805388394,0.437879878387385],[0.385720646703606,0.239893097451638,0.000575460017820722,0.0045201142840198,0.00875223116262325,0.00232851984963184,0.000548603675950233,0.0774921816667691,0.0134042945872997,0.00538431481395258,0.0139438070345552,0.00652327085489671,0.00391347858025163,0.00513499761293239,0.00453720973591854,0.000450647139494753,0.000463600186816279,0.000549757028389152,0.00540133897580829,0.00784472931285646,0.00416734197566052,0.00407420335729166,3.89772052330967e-06,0.00572616535925939,0.000161036753058955,0.00235688057196743,0.000795682973932705,6.20941878083319e-05,0.00320375098042565,0.000147254290441785,0.000283378744783757,0.000728707959222008,2.68579840621333e-05,0.00161725178721585,0.00198355290238321,0.00144987823755146,2.6793706107386e-06,0.000320819153712994,0.00189910413218406,0.0014119361948261,0.000101006318802721,0.000117330663791679,1.3636276208453e-05,0.000384015589530075,0.00182492684415314,0.0158356250582808,0.00776771586384483,0.00671192768683066,0.00976209485846052,1.70151254013515e-05,5.88673406316854e-06,2.5068769063128e-05,0.00028065485861897,0.000305465073519176,4.93356402763104e-05,0.000241745533092786,0.000433140654814033,8.73893363654845e-05,1.54489875102717e-05,0.00192097614287581,4.24862586362151e-05,0.000901214368438744]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pval<\/th>\n      <th>rsq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="factors-extracted-from-rs9557207-related-genes." class="section level4">
<h4>8 factors extracted from rs9557207-related genes.</h4>
<p>We got 4 of 8 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)</p>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_955_ebi2.jpeg" width="80%">
</figure>
<div id="htmlwidget-60a063183f0059ee486d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-60a063183f0059ee486d">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4","factor5","factor6","factor7","factor8"],[1.32533804706935e-123,0.0587662423948408,0.00888687651645884,0.0076110080797729,0.516826067575,0.504039903941539,5.19069979863102e-12,0.998917855822537],[0.567228739809323,0.00533681505329052,0.0102028962796006,0.0106141081554524,0.000629348415391957,0.000668570376402974,0.06884628311635,2.7557448529209e-09]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pval<\/th>\n      <th>rsq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="factors-extracted-from-rs11616269-related-genes." class="section level4">
<h4>4 Factors extracted from rs11616269-related genes.</h4>
<p>We got 3 of 4 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)</p>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_116_ebi2.jpeg" width="80%">
</figure>
<div id="htmlwidget-55a9e6739eae3c0359ef" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-55a9e6739eae3c0359ef">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4"],[2.99340459375602e-25,4.44295185024457e-36,0.0107848070519112,0.208720906706364],[0.14911331071797,0.209922281316041,0.00969120619330795,0.00236465382989312]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pval<\/th>\n      <th>rsq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="factors-cis-genetic-component-of-ebi2-expression" class="section level3">
<h3>Factors ~ cis-genetic component of EBI2 expression</h3>
<p>In this part, we did association test for factors and cis-genetic component of EBI2 expression. To get the cis-genetic component of EBI2 expression, we performed lasso/elastic_net regression for EBI2 expression ~ cis-snp (within 500kb from TSS of EBI2, 3634 variants).</p>
<div id="regression-ebi2-expression-cis-snp." class="section level4">
<h4>Regression EBI2 expression ~ cis-snp.</h4>
<p>The r-squared of lasso regression is 0.0061 and the r-squared of elastic-net is 0.0057 (lambda = lambda.min). Both lasso and elastic-net just selected a few snps when lambda = lambda.min. It seems that the expression of EBI2 are not predictable using cis-snps. So the results in this part are not as good as those in last part.</p>
</div>
<div id="factors-extracted-from-all-idin-genes.-1" class="section level4">
<h4>62 factors extracted from all IDIN genes.</h4>
<p>We got 12 of 62 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)</p>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_all_cisebi2.jpeg" width="80%">
</figure>
<div id="htmlwidget-be48ecea98e77c8c07c0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-be48ecea98e77c8c07c0">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4","factor5","factor6","factor7","factor8","factor9","factor10","factor11","factor12","factor13","factor14","factor15","factor16","factor17","factor18","factor19","factor20","factor21","factor22","factor23","factor24","factor25","factor26","factor27","factor28","factor29","factor30","factor31","factor32","factor33","factor34","factor35","factor36","factor37","factor38","factor39","factor40","factor41","factor42","factor43","factor44","factor45","factor46","factor47","factor48","factor49","factor50","factor51","factor52","factor53","factor54","factor55","factor56","factor57","factor58","factor59","factor60","factor61","factor62"],[0.0286919706856584,0.111022976726801,0.00286828932345899,0.0449020909310475,0.827885931497213,0.143280600706323,0.50653917158617,0.0225028953571892,0.469939190417693,0.212224625980806,0.64915623793646,0.92639090289765,0.168765435837033,0.563569539225472,0.56041352140127,0.563631018777197,0.548201640802356,0.692058897882861,0.538462793861384,0.912917757946641,0.00769600113913638,0.322843242268848,0.215642586096631,0.580369405346956,0.812938865115284,0.388988679289203,0.497908260476948,0.415570016291696,0.407169236809776,0.883484616642117,0.915115753349736,0.711008615027263,0.101702579782925,0.304971219598827,0.501877515134807,0.00270612595830938,0.568231153101654,0.088034941976905,0.93162956266748,0.735522105861325,0.793716777382595,0.337816597814661,0.073619896759015,0.423158293051098,0.125471244001677,0.104849120419531,0.690621840139092,0.00580680571608389,0.0237192993017613,0.267501419035246,0.233560271894847,0.636722339727463,0.592835205979796,0.197923942580682,0.925795273347766,0.775233192129005,0.565144202151793,0.289955403678723,0.807021133892759,0.709782968367194,0.248667065699512,0.230940669994676],[0.00714439000573827,0.00379741231137513,0.0132291728525814,0.00600781370245774,7.08120498049401e-05,0.00320439306512232,0.000660769667219342,0.00776961800084644,0.000781894594595799,0.00232864738899057,0.000310012601198159,1.2786850715775e-05,0.00283324433000647,0.000499592343231794,0.000507719211460081,0.000499434899151637,0.000540001892121775,0.000234976857753117,0.00056672492391883,1.79165224158395e-05,0.0105845997411453,0.00146318736237279,0.00229419439237529,0.000457779435128014,8.38867857984907e-05,0.00111114364849781,0.000687990716679837,0.000992479079060741,0.00102876009681964,3.21760571954919e-05,1.70201121570089e-05,0.000205629138608492,0.00400489990973339,0.00157528075557124,0.00067537313243664,0.0133863149590504,0.00048774771785684,0.00434997529958273,1.10272662334856e-05,0.000170923907083144,0.000102430056282416,0.0013753820565671,0.00478314029539871,0.000960627509527717,0.00351091378243675,0.0039326054806883,0.000237294635930599,0.0113349571797439,0.00763369190918642,0.00184027362263987,0.00212365456913647,0.000334108105938616,0.000428280096881522,0.00248025524722652,1.29952255023704e-05,0.000122136411985009,0.000495570169058488,0.00167623845922079,8.93865415120628e-05,0.000207459802855224,0.00199168630355354,0.00214759000707987]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pval<\/th>\n      <th>rsq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="factors-extracted-from-rs9557207-related-genes.-1" class="section level4">
<h4>8 factors extracted from rs9557207-related genes.</h4>
<p>We got 4 of 8 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)</p>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_955_cisebi2.jpeg" width="80%">
</figure>
<div id="htmlwidget-611b9542e397b11ad7ed" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-611b9542e397b11ad7ed">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4","factor5","factor6","factor7","factor8"],[0.00306770954960689,0.371633341922315,0.0453637354759487,0.989575969450498,0.276706764451199,0.100014039233726,0.265657626374693,0.85743156300551],[0.0130478283902983,0.00119508385313699,0.00598212952497406,2.55719879233259e-07,0.0017710041672291,0.00404470908733273,0.0018545057691298,4.83466736928224e-05]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pval<\/th>\n      <th>rsq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="factors-extracted-from-rs11616269-related-genes.-1" class="section level4">
<h4>4 Factors extracted from rs11616269-related genes.</h4>
<p>We got 3 of 4 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)</p>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_116_cisebi2.jpeg" width="80%">
</figure>
<div id="htmlwidget-4f80e6833b4adeafdb8e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4f80e6833b4adeafdb8e">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4"],[0.146033810888491,0.207558107174582,0.034187210694759,0.149175626439538],[0.00316083292855426,0.00237676165165156,0.00669704511753788,0.00311223962137706]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pval<\/th>\n      <th>rsq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6    knitr_1.28      whisker_0.3-2   magrittr_1.5   
 [5] xtable_1.8-4    R6_2.4.1        stringr_1.4.0   highr_0.8      
 [9] tools_3.6.1     DT_0.13         xfun_0.13       git2r_0.26.1   
[13] htmltools_0.3.6 crosstalk_1.0.0 yaml_2.2.1      digest_0.6.25  
[17] rprojroot_1.3-2 shiny_1.3.2     later_0.8.0     htmlwidgets_1.3
[21] promises_1.0.1  fs_1.3.1        mime_0.9        glue_1.4.1     
[25] evaluate_0.14   rmarkdown_1.13  stringi_1.4.6   compiler_3.6.1 
[29] backports_1.1.7 jsonlite_1.6.1  httpuv_1.5.1   </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
