<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2020-04-02" />

<title>pwy_PLIER</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">pathway_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/pathway_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">pwy_PLIER</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2020-04-02</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-04-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>pathway_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200323code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200323)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200323code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200323)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229pathwayanalysistree40a9578cac8d3178e9e25af9856ec4d22d9e21e5targetblank40a9578a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/pathway_analysis/tree/40a9578cac8d3178e9e25af9856ec4d22d9e21e5" target="_blank">40a9578</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229pathwayanalysistree40a9578cac8d3178e9e25af9856ec4d22d9e21e5targetblank40a9578a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/pathway_analysis/tree/40a9578cac8d3178e9e25af9856ec4d22d9e21e5" target="_blank">40a9578</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/pwy_PLIER.rmd</code>) and HTML (<code>docs/pwy_PLIER.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/40a9578cac8d3178e9e25af9856ec4d22d9e21e5/analysis/pwy_PLIER.rmd" target="_blank">40a9578</a>
</td>
<td>
XSun
</td>
<td>
2020-04-09
</td>
<td>
update heritability and association test
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/pathway_analysis/4cb8c897403beb5a5453c54e349fc4e9fe471bff/docs/pwy_PLIER.html" target="_blank">4cb8c89</a>
</td>
<td>
XSun
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/84481a66050d7975b7705d1d1afbe037c7f57832/analysis/pwy_PLIER.rmd" target="_blank">84481a6</a>
</td>
<td>
XSun
</td>
<td>
2020-04-02
</td>
<td>
update plier
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/pathway_analysis/af07d9b4e88d6990112f1095044827fd992fd631/docs/pwy_PLIER.html" target="_blank">af07d9b</a>
</td>
<td>
XSun
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/b93d36387f3041902c1830baf6e406892cd168a2/analysis/pwy_PLIER.rmd" target="_blank">b93d363</a>
</td>
<td>
XSun
</td>
<td>
2020-04-02
</td>
<td>
update plier
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/pathway_analysis/016fff00d6f48a5c74956af75a09004c842e411a/docs/pwy_PLIER.html" target="_blank">016fff0</a>
</td>
<td>
XSun
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/pathway_analysis/blob/2c1659f2d63f8c4ce6538b07a37fcf7e7a54a6b8/analysis/pwy_PLIER.rmd" target="_blank">2c1659f</a>
</td>
<td>
XSun
</td>
<td>
2020-04-02
</td>
<td>
update plier
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>PLIER(pathway-level information extractor) is a broadly applicable method to accurately infer relevant biological insights, such as variation in cell-type proportion or pathway activity, from global gene expression studies. It approximates the expression pattern of every gene as a linear combination of eigengene-like latent variables (LVs). We calculated heritability of these LVs and performed further analysis.</p>
</div>
<div id="material-and-methods" class="section level2">
<h2>Material and Methods</h2>
<ul>
<li>We used whole blood gene expression data from GTEx.<br />
</li>
<li>Using <a href="https://github.com/wgmao/PLIER">PLIER</a> to reduce the dimension of gene expression matrix and get LVs. We used default settings as PLIER suggested.</li>
<li>Estimate heritability of the LVs in last step using <a href="https://cnsgenomics.com/software/gcta/#GREML">GCTA</a></li>
<li>Perform association test between the LVs with high heritability and all SNPs using Plink. Corrected the results for genotype PCs. We used Plink to compute 10 PCs and used them as corvariates in association test.</li>
</ul>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<div id="overview" class="section level3">
<h3>Overview</h3>
<ul>
<li>The gene expression data from GTEx using Ensemnl ID. In order to use PLIER R package, we need to convert Ensemnl ID to HGNC Gene Symbol. There are 20315 genes in GTEx data. After conversion, 17143 left(not all Ensemnl IDs have a corresponding HGNC Gene Symbol). But only 5641 of 17143 were used in PLIER.</li>
<li>We used CanonicalPathways in PLIER R package, which contains 545 curated pathways from Reactome and KEGG.</li>
<li>We obtained 90 LVs from PLIER, 26 of them have AUC &gt; 0.70(see figure below).</li>
</ul>
</div>
<div id="lvs-with-auc0.70" class="section level3">
<h3>LVs with AUC&gt;0.70</h3>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/Umat_wb.jpeg" width="80%">
</figure>
<p>x-axis shows the name of LVs and y-axis shows the pathway names related to each LVs.</p>
</div>
<div id="top-genes-and-their-pathway-assiciations." class="section level3">
<h3>Top genes and their pathway assiciations.</h3>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/topgenes_wb.jpeg" width="80%">
</figure>
<p>Only LVs with pathway association are plotted.</p>
</div>
<div id="heritability-of-lvs" class="section level3">
<h3>Heritability of LVs</h3>
<p>We computed the heritability of all LVs. The following figure shows the distribution of the heritability.</p>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/hist_H_lv.jpeg" width="50%">
</figure>
<p>For more details, we list the heritability of 26 LVs with AUC &gt; 0.7 below. We just list the top 3 pathways related to each LV, corresponding to the figure in ‘LVs with AUC&gt;0.70’ part.</p>
<table>
<colgroup>
<col width="1%" />
<col width="4%" />
<col width="3%" />
<col width="8%" />
<col width="8%" />
<col width="7%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th>LVs</th>
<th>V(G)/Vlvs</th>
<th>SE</th>
<th>zscore</th>
<th>pval</th>
<th>LV_AUC</th>
<th>pwys</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>60</td>
<td>0.494548</td>
<td>0.171469</td>
<td>2.88418314680788</td>
<td>0.00196215225033847</td>
<td>0.726392931195191</td>
<td>1.REACTOME_GENERIC_TRANSCRIPTION_PATHWAY,</td>
</tr>
<tr class="even">
<td>48</td>
<td>0.493311</td>
<td>0.202114</td>
<td>2.44075620689314</td>
<td>0.00732827268754888</td>
<td>0.996043956043956</td>
<td>1.REACTOME_MITOCHONDRIAL_PROTEIN_IMPORT,2.REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES,3.KEGG_OXIDATIVE_PHOSPHORYLATION,</td>
</tr>
<tr class="odd">
<td>36</td>
<td>0.400558</td>
<td>0.19515</td>
<td>2.05256469382526</td>
<td>0.0200574073382793</td>
<td>0.897281661412943</td>
<td>1.REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA,2.REACTOME_MRNA_PROCESSING,3.REACTOME_MRNA_SPLICING,</td>
</tr>
<tr class="even">
<td>71</td>
<td>0.397832</td>
<td>0.194249</td>
<td>2.04805172742202</td>
<td>0.0202774655833539</td>
<td>0.997166565472576</td>
<td>1.REACTOME_E2F_MEDIATED_REGULATION_OF_DNA_REPLICATION,2.PID_ATR_PATHWAY,3.KEGG_P53_SIGNALING_PATHWAY,</td>
</tr>
<tr class="odd">
<td>81</td>
<td>0.256039</td>
<td>0.164415</td>
<td>1.55727275491896</td>
<td>0.0597028705626842</td>
<td>0.797549019607843</td>
<td>1.PID_ERBB1_DOWNSTREAM_PATHWAY,2.REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA,3.REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM,</td>
</tr>
<tr class="even">
<td>1</td>
<td>0.239428</td>
<td>0.184462</td>
<td>1.2979800717763</td>
<td>0.0971470914853219</td>
<td>0.876725304465494</td>
<td>1.KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION,2.KEGG_CELL_ADHESION_MOLECULES_CAMS,3.REACTOME_IMMUNOREGULATORY_INTERACTIONS_BETWEEN_A_LYMPHOID_AND_A_NON_LYMPHOID_CELL,</td>
</tr>
<tr class="odd">
<td>22</td>
<td>0.223824</td>
<td>0.165512</td>
<td>1.3523128232394</td>
<td>0.0881376319708236</td>
<td>0.964356763925729</td>
<td>1.KEGG_CELL_ADHESION_MOLECULES_CAMS,2.KEGG_INTESTINAL_IMMUNE_NETWORK_FOR_IGA_PRODUCTION,3.KEGG_PRIMARY_IMMUNODEFICIENCY,</td>
</tr>
<tr class="even">
<td>76</td>
<td>0.201392</td>
<td>0.162757</td>
<td>1.23737842304786</td>
<td>0.107973312713352</td>
<td>0.79299641862316</td>
<td>1.KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION,2.REACTOME_PEPTIDE_LIGAND_BINDING_RECEPTORS,3.KEGG_HEMATOPOIETIC_CELL_LINEAGE,</td>
</tr>
<tr class="odd">
<td>82</td>
<td>0.195819</td>
<td>0.171871</td>
<td>1.13933706093524</td>
<td>0.127281298120168</td>
<td>0.864578313253012</td>
<td>1.REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA,2.REACTOME_SIGNALING_BY_WNT,3.REACTOME_TRANSPORT_OF_MATURE_MRNA_DERIVED_FROM_AN_INTRONLESS_TRANSCRIPT,</td>
</tr>
<tr class="even">
<td>79</td>
<td>0.161897</td>
<td>0.106225</td>
<td>1.52409508119558</td>
<td>0.0637424806329094</td>
<td>0.856885923101366</td>
<td>1.REACTOME_DIABETES_PATHWAYS,2.REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES,3.REACTOME_UNFOLDED_PROTEIN_RESPONSE,</td>
</tr>
<tr class="odd">
<td>23</td>
<td>0.126631</td>
<td>0.090764</td>
<td>1.39516768762946</td>
<td>0.0814826392421427</td>
<td>0.993530997304582</td>
<td>1.KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION,2.REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES,3.KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450,</td>
</tr>
<tr class="even">
<td>57</td>
<td>0.10431</td>
<td>0.136525</td>
<td>0.764035890862479</td>
<td>0.222422927303134</td>
<td>0.906291320110898</td>
<td>1.KEGG_VASCULAR_SMOOTH_MUSCLE_CONTRACTION,2.REACTOME_SIGNALING_BY_PDGF,3.KEGG_ECM_RECEPTOR_INTERACTION,</td>
</tr>
<tr class="odd">
<td>89</td>
<td>0.09778</td>
<td>0.109456</td>
<td>0.893326998976758</td>
<td>0.185841041767189</td>
<td>0.774674181675942</td>
<td>1.PID_MTOR_4PATHWAY,2.REACTOME_GLUCOSE_METABOLISM,3.PID_HIF1_TFPATHWAY,</td>
</tr>
<tr class="even">
<td>39</td>
<td>0.097511</td>
<td>0.089272</td>
<td>1.0922909758939</td>
<td>0.137352612772346</td>
<td>0.757300275482094</td>
<td>1.REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES,2.KEGG_COMPLEMENT_AND_COAGULATION_CASCADES,</td>
</tr>
<tr class="odd">
<td>51</td>
<td>0.096981</td>
<td>0.132827</td>
<td>0.730130169317985</td>
<td>0.232655310879605</td>
<td>0.943730044816778</td>
<td>1.REACTOME_ORC1_REMOVAL_FROM_CHROMATIN,2.KEGG_LYSOSOME,3.KEGG_PROTEASOME,</td>
</tr>
<tr class="even">
<td>15</td>
<td>0.079205</td>
<td>0.125819</td>
<td>0.629515415000914</td>
<td>0.264505840043043</td>
<td>0.992077369773115</td>
<td>1.REACTOME_NEGATIVE_REGULATORS_OF_RIG_I_MDA5_SIGNALING,2.REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM,3.KEGG_CYTOSOLIC_DNA_SENSING_PATHWAY,</td>
</tr>
<tr class="odd">
<td>40</td>
<td>0.073455</td>
<td>0.097723</td>
<td>0.751665421650993</td>
<td>0.226126144895035</td>
<td>0.953350144092219</td>
<td>1.REACTOME_MUSCLE_CONTRACTION,2.KEGG_DILATED_CARDIOMYOPATHY,</td>
</tr>
<tr class="even">
<td>78</td>
<td>0.039485</td>
<td>0.081458</td>
<td>0.48472832625402</td>
<td>0.313934535218113</td>
<td>0.922911374541349</td>
<td>1.REACTOME_MITOCHONDRIAL_PROTEIN_IMPORT,2.REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES,3.MIPS_NOP56P_ASSOCIATED_PRE_RRNA_COMPLEX,</td>
</tr>
<tr class="odd">
<td>37</td>
<td>0.027415</td>
<td>0.049898</td>
<td>0.549420818469678</td>
<td>0.291358345163622</td>
<td>0.866641901931649</td>
<td>1.REACTOME_GLUCOSE_METABOLISM,2.REACTOME_MUSCLE_CONTRACTION,3.KEGG_ARGININE_AND_PROLINE_METABOLISM,</td>
</tr>
<tr class="even">
<td>3</td>
<td>1e-06</td>
<td>0.053586</td>
<td>1.86615907139925e-05</td>
<td>0.499992555102445</td>
<td>0.700039611804318</td>
<td>1.KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION,2.PID_NFAT_TFPATHWAY,3.REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM,</td>
</tr>
<tr class="odd">
<td>6</td>
<td>1e-06</td>
<td>0.035406</td>
<td>2.82438004857934e-05</td>
<td>0.499988732353828</td>
<td>0.828101181402439</td>
<td>1.REACTOME_GLYCOSAMINOGLYCAN_METABOLISM,2.KEGG_LYSOSOME,3.REACTOME_INNATE_IMMUNE_SYSTEM,</td>
</tr>
<tr class="even">
<td>9</td>
<td>1e-06</td>
<td>0.120094</td>
<td>8.3268106649791e-06</td>
<td>0.499996678083165</td>
<td>0.718790926099158</td>
<td>1.KEGG_CELL_ADHESION_MOLECULES_CAMS,2.KEGG_LYSOSOME,3.KEGG_INTESTINAL_IMMUNE_NETWORK_FOR_IGA_PRODUCTION,</td>
</tr>
<tr class="odd">
<td>31</td>
<td>1e-06</td>
<td>0.036497</td>
<td>2.73995122886813e-05</td>
<td>0.499989069176087</td>
<td>0.765705512811463</td>
<td>1.REACTOME_TRANSMISSION_ACROSS_CHEMICAL_SYNAPSES,2.REACTOME_NEURONAL_SYSTEM,</td>
</tr>
<tr class="even">
<td>62</td>
<td>1e-06</td>
<td>0.042027</td>
<td>2.37942275204035e-05</td>
<td>0.499990507476614</td>
<td>0.996922513280821</td>
<td>1.REACTOME_3_UTR_MEDIATED_TRANSLATIONAL_REGULATION,2.REACTOME_FORMATION_OF_THE_TERNARY_COMPLEX_AND_SUBSEQUENTLY_THE_43S_COMPLEX,3.KEGG_RIBOSOME,</td>
</tr>
<tr class="odd">
<td>66</td>
<td>1e-06</td>
<td>0.169204</td>
<td>5.91002576771235e-06</td>
<td>0.499997642240843</td>
<td>0.851892358774173</td>
<td>1.KEGG_OXIDATIVE_PHOSPHORYLATION,2.REACTOME_NONSENSE_MEDIATED_DECAY_ENHANCED_BY_THE_EXON_JUNCTION_COMPLEX,3.MIPS_NOP56P_ASSOCIATED_PRE_RRNA_COMPLEX,</td>
</tr>
<tr class="even">
<td>72</td>
<td>1e-06</td>
<td>0.043159</td>
<td>2.31701383257258e-05</td>
<td>0.49999075645218</td>
<td>1</td>
<td>1.REACTOME_3_UTR_MEDIATED_TRANSLATIONAL_REGULATION,2.KEGG_OXIDATIVE_PHOSPHORYLATION,3.REACTOME_NONSENSE_MEDIATED_DECAY_ENHANCED_BY_THE_EXON_JUNCTION_COMPLEX,</td>
</tr>
</tbody>
</table>
</div>
<div id="association-test-results" class="section level3">
<h3>Association test results</h3>
<p>Since SNPs in MHC has p &lt; 1e-100, We show the results after removing MHC.</p>
<div id="results-for-lv60" class="section level4">
<h4>Results for LV60</h4>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/qp_unif_lv60.jpeg" width="50%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/man_lv60.jpeg" width="50%">
</figure>
</div>
<div id="results-for-lv48" class="section level4">
<h4>Results for LV48</h4>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/qp_unif_lv48.jpeg" width="50%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/man_lv48.jpeg" width="50%">
</figure>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4      rprojroot_1.3-2 digest_0.6.25   later_0.8.0    
 [5] R6_2.4.1        backports_1.1.5 git2r_0.26.1    magrittr_1.5   
 [9] evaluate_0.14   stringi_1.4.6   fs_1.3.1        promises_1.0.1 
[13] whisker_0.3-2   rmarkdown_1.13  tools_3.6.1     stringr_1.4.0  
[17] glue_1.3.2      httpuv_1.5.1    xfun_0.12       yaml_2.2.1     
[21] compiler_3.6.1  htmltools_0.3.6 knitr_1.28     </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
