---
title: "factor_flash"
author: "XSun"
date: "2020-05-27"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this part, we used ['flashr'](https://github.com/stephenslab/flashr), methods for matrix factorization based on [Empirical Bayes Matrix Factorization](https://arxiv.org/abs/1802.06931), to extract factors from IDIN gene expression matrix and part of the IDIN gene expression matrix.We also performed factor analysis find if any factor associates with the two SNPs ( [rs9557207](https://www.ncbi.nlm.nih.gov/snp/rs9557207) and [rs11616269](https://www.ncbi.nlm.nih.gov/snp/rs11616269)) and EBI2 expression data.

## Material and Methods
1. Data
    - All genes in IDIN(same with the genes in [IDIN-PLIER](https://xsun1229.github.io/pathway_analysis/IDIN.html))
    - Part of the IDIN genes (genes associated with rs9557207 and rs11616269, listed by *Bayesian detection of expression quantitative trait [Genetics, 2011]*, see the table below,  evidence of associations :BF > 10)
    - EBI2 expression data and genotype of variants are from GTEx data.

**rs11616269**
```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/gene_BF_116.rdata")
DT::datatable(gene_116, options = list(pageLength =20))

```

**rs9557207**
```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/gene_BF_955.rdata")
DT::datatable(gene_955, options = list(pageLength =20))

```


2. We used workflow showed in ['flashr'](https://github.com/stephenslab/flashr) to extract factors. 

```{r eval=FALSE, echo=T}
f = flash(Y)    ### Y is the gene expression matrix.
ldf = flash_get_ldf(f)
```

3. Then, we used the backfitting algorithm to improve the fit (but it takes longer).

```{r eval=FALSE, echo=T}
f.b = flash(Y, f_init = f, backfit=TRUE, greedy=FALSE)
ldf.b = flash_get_ldf(f.b)
```

4. Perform association tests between the facotrs and two SNPs/EBI2 expression/cis-genetic component of EBI2 expression. 

5. Besides, I performed association tests for the SNP-related gene expression and snp to find if they are associated with each other (single gene expression ~ single snp).

## Results

We obtained 62 factors for all genes in IDIN using FLASH. For genes associated with rs9557207, we got 8 factors and for genes associated with rs11616269, we got 4.

### Factors ~ SNPs.

#### All genes in IDIN

We performed linear regression for factors(extracted from all genes in IDIN) ~ snp.

```{r eval=FALSE, echo=T}
  ### k indicates the kth factors
  y <- as.numeric(factors[,k])   ## kth factors
  x <- as.numeric(geno_2snp[,1])   ## genotype of rs11616269
  
  pval_snp1[k] <- summary(lm(y ~ x))$coefficients[2,4]  ## pval of F-statistics in y ~ x regression
  rsq_snp1[k] <- summary(lm(y ~ x))$r.squared    ## r-squared of the regression
```

We made a histogram to show the pvalue distribution. 

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/rs11616269_pval_flash_b.jpeg" width="80%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/rs9557207_pval_flash_b.jpeg" width="80%">
</figure>

The table below shows the detailed results of the linear regression. 

Notations: **pval** is the pvalue of F-statistics in regression. **rsq** is the r-squared of the regression.

You can sort the table by clicking the arrow by the header of each column.


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/summary_assoc_2snp_flash_b.rdata")
rownames(summary) <- paste0("factor", seq(1, nrow(summary), by=1))
DT::datatable(summary, options = list(pageLength =20))
```

#### Genes associated with rs9557207

We performed linear regression for factors ~ snp. We list the p-values and r-squared of each regression here. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/summary_assoc_955_flash_b.rdata")
rownames(summary) <- paste0("factor", seq(1, nrow(summary), by=1))
DT::datatable(summary, options = list(pageLength =10))
```


#### Genes associated with rs11616269

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/summary_assoc_116_flash_b.rdata")
rownames(summary) <- paste0("factor", seq(1, nrow(summary), by=1))
DT::datatable(summary, options = list(pageLength =10))
```

### Gene expression ~ SNPs

*It seems that the two snps are not associated with the gene expression(genes are reported in [Bayesian detection of expression quantitative trait (Genetics, 2011)]. So I'm considering the factor analysis results are reasonable.*


- rs9557207

```{r echo=FALSE}
summary <- read.csv("/project2/xinhe/xsun/website/pathway_analysis/output/rs9557207_gene_snp.csv", header = T)
DT::datatable(summary, options = list(pageLength =10))
```

- rs11616269

```{r echo=FALSE}
summary <- read.csv("/project2/xinhe/xsun/website/pathway_analysis/output/rs11616269_gene_snp.csv", header = T)
DT::datatable(summary, options = list(pageLength =10))
```


### Factors ~ EBI2 expression

We made histograms to show the pvalue distribution and we also made a table to show the detailed information for each factor.

#### 62 factors extracted from all IDIN genes.

We got 12 of 62 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_allgene_ebi2.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/p_r_all_ebi2.rdata")
DT::datatable(summary, options = list(pageLength =10))
```


#### 8 factors extracted from rs9557207-related genes.

We got 4 of 8 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_955_ebi2.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/p_r_955_ebi2.rdata")
DT::datatable(summary, options = list(pageLength =10))
```

#### 4 Factors extracted from rs11616269-related genes.

We got 3 of 4 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_116_ebi2.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/p_r_116_ebi2.rdata")
DT::datatable(summary, options = list(pageLength =10))
```

### Factors ~ cis-genetic component of EBI2 expression

In this part, we did association test for factors and cis-genetic component of EBI2 expression. To get the cis-genetic component of EBI2 expression, we performed lasso/elastic_net regression for EBI2 expression ~ cis-snp (within 500kb from TSS of EBI2, 3634 variants). 

#### Regression EBI2 expression ~ cis-snp.

The r-squared of lasso regression is 0.0061 and the r-squared of elastic-net is 0.0057 (lambda = lambda.min). Both lasso and elastic-net just selected a few snps when lambda = lambda.min. It seems that the expression of EBI2 are not predictable using cis-snps. So the results in this part are not as good as those in last part.

Lasso

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/ebi2_mse_l.jpeg" width="80%">
</figure>

Elastic-net

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/ebi2_mse_l.jpeg" width="80%">
</figure>

Since the results of lasso and elastic-net are not quite different, we just attatch the results of lasso regression in the following part.

#### 62 factors extracted from all IDIN genes.

We got 8 of 62 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_all_cisebi2.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/p_r_all_cisebi2.rdata")
DT::datatable(summary, options = list(pageLength =10))
```


#### 8 factors extracted from rs9557207-related genes.

We got 2 of 8 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_955_cisebi2.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/p_r_955_cisebi2.rdata")
DT::datatable(summary, options = list(pageLength =10))
```

#### 4 Factors extracted from rs11616269-related genes.

We got 1 of 4 factors associated with EBI2 expression (pvalues of F-statistics in linear regression are less than 0.05)

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/pval_116_cisebi2.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/p_r_116_cisebi2.rdata")
DT::datatable(summary, options = list(pageLength =10))
```

