---
title: "INID"
author: "XSun"
date: "2020-05-24"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In [Bayesian detection of expression quantitative trait [Genetics, 2011]](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3241411/), authors used HESS to analyze the larger IDIN(IRF7-driven inflammatory network) and found two SNPs, [rs9557207](https://www.ncbi.nlm.nih.gov/snp/rs9557207) and [rs11616269](https://www.ncbi.nlm.nih.gov/snp/rs11616269),which are detected as hotspots for the IDIN expression. 

Here, we perform factor analysis using IDIN genes in whole blood data (GTEx) to find if any factor associates with the two SNPs. 

## Material and Methods

1. Genes we used here come from IRF7-driven inflammatory network (IDIN).
Data source: Table S5 from *Heinig M., Petretto E., Wallace C., Bottolo L., Rotival M., et al. , 2010. A trans-acting locus regulates an anti-viral expression network and type 1 diabetes risk. Nature 467: 460464*

```{r echo=FALSE}
IDIN <- read.csv("/project2/xinhe/xsun/website/pathway_analysis/output/IDIN.csv", header = T)
DT::datatable(IDIN, options = list(pageLength =5))

```

2. Using [PLIER](https://github.com/wgmao/PLIER) to reduce the dimension of gene expression matrix and get LVs. We used default settings as PLIER suggested. 

3. Perform association test between the LVs and two SNPs.

## Results

### Overview

1. There are 524 genes(ensembl.id) in IDIN. 483 of them have expression data in whole blood data (GTEx). After converting ensembl.id into hgnc_symbol, 480 genes left. We used the gene expression matrix to of these genes to extract PLIER factors (LVs). 

2.  We used all pathway data provided in PLIER R package, which contains 6611 curated pathways. Among 480 genes, 435 genes were used in factor extraction.

    (1) bloodCellMarkersIRISDMAP - 
    IRIS: Abbas, Alexander R., et al. "Deconvolution of blood microarray data identifies cellular activation patterns in systemic lupus erythematosus." PloS one 4.7 (2009).
    DMAP: Novershtern, Noa, et al. "Densely interconnected transcriptional circuits control cell states in human hematopoiesis." Cell 144.2 (2011): 296-309.
    (2) canonicalPathways  - canonical pathway from MSigDB
    (3) chemgenPathways - chemical and genetic perturbations from MSigDB
    (4) immunePathways  - Immunologic signatures from MSigDB
    (5) oncogenicPathways - oncogeneic signatures gene sets from MsigDB
    (6)svmMarkers - Newman, Aaron M., et al. "Robust enumeration of cell subsets from tissue expression profiles." Nature methods 12.5 (2015): 453-457.
    (7) xCell - Aran, Dvir, Zicheng Hu, and Atul J. Butte. "xCell: digitally portraying the tissue cellular heterogeneity landscape." Genome biology 18.1 (2017): 220.

3. We obtained 86 LVs from PLIER, 29 of them have AUC > 0.70(see figure below)

### LV plots

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/Umat_wb_idin.jpeg" width="80%">
</figure>

x-axis shows the name of LVs and y-axis shows the pathway names related to each LVs.

### Association test

We performed linear regression for LVs ~ snp. 

```{r eval=FALSE, echo=T}
  ### k indicates the kth LV
  y <- as.numeric(LVs[,k])   ## kth LV
  x <- as.numeric(geno_2snp[,1])   ## genotype of rs11616269
  
  pval_snp1[k] <- summary(lm(y ~ x))$coefficients[2,4]  ## pval of F-statistics in y ~ x regression
  rsq_snp1[k] <- summary(lm(y ~ x))$r.squared    ## r-squared of the regression
```

We made a histogram to show the pvalue distribution. 

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/rs11616269_pval.jpeg" width="80%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/rs9557207_pval.jpeg" width="80%">
</figure>

The table below shows the detailed results of the linear regression. 

Notations: **LV_AUC** is an indicator to show the confidence that LV will associate with some pathways. AUV > 0.7 means high confidence. **pval** is the pvalue of F-statistics in regression. **rsq** is the r-squared of the regression.

You can sort the table by clicking the arrow by the header of each column.

```{r eval=FALSE, echo=T}
LV_AUC <- apply(plierResult$Uauc, 2, max)  ### plierResult is the results of factor extraction.
```

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/summary_assoc_2snp.rdata")
DT::datatable(summary, options = list(pageLength =20))

```

