---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# Pathway analysis

- KEGG 

    KEGG is a collection of databases dealing with genomes, biological pathways, diseases, drugs, and chemical substances. For each pathway of human, we inferred the PCs and PEER factors of it. Then computed the heritabilities of factors and made a summary of the heritabilities. [For more details](https://xsun1229.github.io/pathway_analysis/analy.html)

- pwy-plier

    PLIER(pathway-level information extractor) is a broadly applicable method to accurately infer relevant biological insights, such as variation in cell-type proportion or pathway activity, from global gene expression studies. It approximates the expression pattern of every gene as a linear combination of eigengene-like latent variables (LVs). We calculated heritability of these LVs and performed further analysis. [For more details](https://xsun1229.github.io/pathway_analysis/pwy_PLIER.html)

- pwy-plier-nochrx

    In last part, we extracted LVs from gene expression matrix which includes chrX, but it maybe more accurate to exclude chrX. So, we performed analysis for the data excluding chrX. [For more details](https://xsun1229.github.io/pathway_analysis/pwy-plier-nochrx.html)
    
- pwy-plier-nochrx-allpathway
    
    In this part, the data and workflow are the same with [pwy-plier-nochrx](https://xsun1229.github.io/pathway_analysis/pwy-plier-nochrx.html). In last part, we extracted LVs using PLIER. But we just included the pathway dataset that containing KEGG pathway. In this part, we get all the pathways provided in 'PLIER' R package included to find if we can get better results. [For more details](https://xsun1229.github.io/pathway_analysis/pwy-plier-nochrx-allpath.html)

# Factor-TWAS

We have extract factors(LVs) related to pathways from gene expression data and calculated their heritability in former analysis. We plan to use TWAS to find the association between diseases and pathways/factors. TWAS analyse can be divided into training and prediction part. 

## Training part

1. In this part, we will use lasso and elastic_net regression to train a prediction model. The dependent variables here are LVs and the independent variables are genotype of certain variants. [For more details](https://xsun1229.github.io/pathway_analysis/train_model.html)

2. In this part, the workflow is the same with [1.](https://xsun1229.github.io/pathway_analysis/train_model.html). For the data part, the genotype data are the same too, we just change the phenotype data. The phenotype data are the LVs obtained from [pwy-plier-nochrx-allpath](https://xsun1229.github.io/pathway_analysis/pwy-plier-nochrx-allpath.html). [For more details](https://xsun1229.github.io/pathway_analysis/train_model_allpath.html)


3. We performed same analysis with [1](https://xsun1229.github.io/pathway_analysis/train_model.html). But the data are kind of different. After dividing the data into training part and testing part, we perfpormed LDpruning for the training data using Plink. The window size of ldpruning is 50kb, the step size is 5kb, the VIF is 5 (1/(1-r2), r2=0.8). After ldpruning, the average number of variants in training data decreases from 103,809 to 31,834. [For more details](https://xsun1229.github.io/pathway_analysis/train_model_ldpruning.html)

4. We also trained a model for the fifth pc of T1B pathway.  [For more details](https://xsun1229.github.io/pathway_analysis/T1B.html)

## Prediction part

# Factor analysis

- IDIN-PLIER

In *Bayesian detection of expression quantitative trait [Genetics, 2011]*, authors used HESS to analyze the larger IDIN(IRF7-driven inflammatory network) and found two SNPs, rs9557207 and rs11616269,which are detected as hotspots for the IDIN expression. 

Here, we perform factor analysis using IDIN genes in whole blood data to find if any factor associates with the two SNPs.  [For more details](https://xsun1229.github.io/pathway_analysis/IDIN.html)

- IDIN-FLASH

In this part, we used 'flashr', methods for matrix factorization based on [Empirical Bayes Matrix Factorization](https://arxiv.org/abs/1802.06931), to extract factors from IDIN gene expression matrix (same with the matrix in [IDIN-PLIER](https://xsun1229.github.io/pathway_analysis/IDIN.html)).We also performed factor analysis find if any factor associates with the two SNPs (rs9557207, rs11616269). [For more details](https://xsun1229.github.io/pathway_analysis/factor_flash.html)


Welcome to my research website.
