---
title: "train_flash_all"
author: "XSun"
date: "2020-06-15"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

We used 'flashr' package to extract factors from GTEx gene expression data and tried to train a prediction model for these factors ~ variant genotype. The workflow here can be found in 'Material and Method' part of  [PLIER 1.](https://xsun1229.github.io/pathway_analysis/train_model.html). [For more details](https://xsun1229.github.io/pathway_analysis/train_flash_all.html)

## Material and Methods

1. Extracting factors. 
    
    - ['flashr'](https://github.com/stephenslab/flashr) was adopted to extract factors.
    
    - We used 670 samples and expression data of 10403209 genes from GTEx data. 
    
    - The expression matrix is too large to perform backfitting algorithm to improve the fit. The program has been running for more than thirty hours and has not ended, and the use of parallel computing has not shortened the time. So I just used factors from
    
```{r eval=FALSE, echo=T}
f = flash(Y)    ### Y is the gene expression matrix.
ldf = flash_get_ldf(f)
```

2. The overal training workflow are the same with [train_model](https://xsun1229.github.io/pathway_analysis/train_model.html).

    - We splited the data set into training(80%, 536 samples) and test(20%, 134 samples) set.

    - In the training data, performed association test between the 100 factors and all SNPs using Plink. We corrected the results for genotype PCs (We used Plink to compute 10 PCs and used them as corvariates in association test).

    - From the results of last step, we selected all SNPs with p < 1e-02. For each factor, we can get a set of around 100,000 variants.

    - Trained the prediction models using SNPs from last step in the training data and evaluated the performance in the testing data. (Using glmnet package)

    - For factors with relatively low pval_test_l, we did GWAS analysis employing all samples provided by GTEx. In the association tests, first 10 genotype pcs were included as covariants.


## Results 

### Heritability of factors

We obtained 100 factors from GTEx gene expression data and made a histograme to show the distribution of the heritability of factors.

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/hist_H_lv_flash_all.jpeg" width="80%">
</figure>

### Summary table for prediction model

The table below shows the heritability of factorss and performance of the training models. The table has been sorted by Heritability from small to large.

Some notations to explain: **rsq_min** is the r-squared when lambda=lambda.min, **ncoef_nzero_min** is the number of non-zero coefficients when lambda=lambda.min, **rsq_test** is the r-squared in testing set (obtained from lm(factor.obs ~ factor.pred)) ,**pval_test** is the pvalue of F-statistics in testing set. When the suffix is '_1se', the notations mean lambda=lambda.1se. Suffix '_l' means lasso regression and '_e' means elastic_net


```{r eval=FALSE, echo=T}
  ###Lasso regression. 'k' represents the kth factor.
    fit_lasso <- cv.glmnet(x, y, alpha =1, foldid = foldid)   ##'foldid = foldid' can fix the data partition in cross validation, making the codes reproducible
    fit_r_train_l[k]  <- fit_lasso$glmnet.fit$dev.ratio[which(fit_lasso$glmnet.fit$lambda == fit_lasso$lambda.min)]  ###r2 when lambda = lambda.min
    predict_l <- predict(fit_lasso, nx, s = fit_lasso$lambda.min )
    
    fit_linear <- lm(y_test ~ predict_l)
    pval_test_l[i] <- summary(fit_linear)$coefficients[,4]
    rsq_test_l[i] <- summary(fit_linear)$r.squared

```

In this table, there are several NA value. These NA values come from linear regression(lm(y_test ~ predict_l)). It means the variant in regression is linearly with the intercept. After checking the predict_l, we found the predict_l for all samples are the same.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/summary_H_le_sorted_flash_all.rdata")
rownames(sum_sorted) <- c()
DT::datatable(sum_sorted, options = list(pageLength =20))
```


### Factors with relatively low pval_test_l

By sorting the pval_test_l in the table above, we can get several factors that have very low pvalues on testing set (factor 95,20,69,67). We made scatter plots of predicted factors vs. observed factors for them. Besides, we did GWAS analysis.

The red line in manhattan plot means p-value is 5e-8. 

#### factor95

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV95_flash_all.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/qq_unif_lv95.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/man_lv95.jpeg" width="200%">
</figure>

#### factor20

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV20_flash_all.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/qq_unif_lv20.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/man_lv20.jpeg" width="200%">
</figure>

#### factor69

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV69_flash_all.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/qq_unif_lv69.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/man_lv69.jpeg" width="200%">
</figure>

#### factor67
  
<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV67_flash_all.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/qq_unif_lv67.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/man_lv67.jpeg" width="200%">
</figure>
