---
title: "train_model_allpath"
author: "XSun"
date: "2020-05-25"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this part, the workflow is the same with [train_model](https://xsun1229.github.io/pathway_analysis/train_model.html). For the data part, the genotype data are the same too, we just change the phenotype data. The phenotype data are the LVs obtained from [pwy-plier-nochrx-allpath](https://xsun1229.github.io/pathway_analysis/pwy-plier-nochrx-allpath.html)


## Results

### Heritability vs R-squared

This figure shows the prediction performance (R2 of Lasso/Elastic_net regression on training set in red) in comparison to LV heritability estimates (black; standard errors in gray). Performance was assessed using tenfold cross-validation in the GTEx whole-blood cohort (on training set, n = 536) with the elastic net and lasso models.
  
<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/devide_snp001_le_allpath.jpeg" width="150%">
</figure>

## Debugging

### Table
The table below shows the heritability of LVs and performance of the training models. The table has been sorted by Heritability from small to large.

Some notations to explain: **rsq_min** is the r-squared when lambda=lambda.min, **ncoef_nzero_min** is the number of non-zero coefficients when lambda=lambda.min, **rsq_test** is the r-squared in testing set (obtained from lm(lv.obs ~ lv.pred)) ,**pval_test** is the pvalue of F-statisticsin testing test. When the suffix is '_1se', the notations mean lambda=lambda.1se. Suffix '_l' means lasso regression and '_e' means elastic_net


```{r eval=FALSE, echo=T}
  ###Lasso regression. 'k' represents the kth LV.
    fit_lasso <- cv.glmnet(x, y, alpha =1, foldid = foldid)   ##'foldid = foldid' can fix the data partition in cross validation, making the codes reproducible
    fit_r_train_l[k]  <- fit_lasso$glmnet.fit$dev.ratio[which(fit_lasso$glmnet.fit$lambda == fit_lasso$lambda.min)]  ###r2 when lambda = lambda.min
    predict_l <- predict(fit_lasso, nx, s = fit_lasso$lambda.min )
    
    fit_linear <- lm(y_test ~ predict_l)
    pval_test_l[i] <- summary(fit_linear)$coefficients[,4]
    rsq_test_l[i] <- summary(fit_linear)$r.squared

```

  
Some LVs like LV6, have pretty low heritability but have very high r-squared. In such LVs, the training model always selected many variants(ncoef_nzero). 
Other LVs like LV76, LV89 have relatively high heritability but very low r-squared.In such LVs, the training model always selected few variants(ncoef_nzero). 


In this table, there are several NA value. These NA values come from linear regression(lm(y_test ~ predict_l)). It means the variant in regression is linearly with the intercept. After checking the predict_l, we found the predict_l for all samples are the same.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/summary_H_le_sorted_allpath.rdata")
rownames(sum_sorted) <- c()
DT::datatable(sum_sorted, options = list(pageLength =20))
```




### lv6 and lv89

Taking lv6(low heritability but high r2) and lv89(high heritability but low r2) as examples, we made sveral plots to show the models. Since the lasso and elastic_net models were not quite different, we just post the plots from lasso regression here.

We first plot the change of mean squared errors and coefficients when training the model using lasso. We also plot the predicted lv values vs observed lv values. 

#### LV6

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV6_mse_allpath.jpeg" width="80%">
</figure>
<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV6_coef_allpath.jpeg" width="80%">
</figure>
<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV6_testpred_allpath.jpeg" width="80%">
</figure>

#### LV89
<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV89_mse_allpath.jpeg" width="80%">
</figure>
<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV89_coef_allpath.jpeg" width="80%">
</figure>
<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV89_testpred_allpath.jpeg" width="80%">
</figure>

For LV89, we performned more analysis to detect the reason why it has relatively high heritability but low r-squared in lasso.

##### Variants with non-zero coefficients

We got 12 variants with non-zero lasso coefficients. For these variants, we performed single snp linear regression (LV89 ~ genotype of a snp) and made conparision in both training set and testing set.

Notations: **coefficients_lasso** is the coefficients from the lasso model. **pval_lm_traindata** is  the p-value of F-statistics from the linear regression using the training set data. **rsq_lm_traindata** is the r-squared of the linear regression using training set data. Suffix "_testdata" means the results using testing set. 


```{r eval=FALSE, echo=T}
### i is the ith variants
  fit_linear[[i]] <- lm(y ~ x[,i])   ###y is T1B_PC5 and  x[,i] is genotype of a variant.
  pval_snp[i] <- summary(fit_linear[[i]])$coefficients[2,4] ### p-value of F-statistics 
  rsq_snp[i] <- summary(fit_linear[[i]])$r.squared   ###r-squared of the linear regression

```

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/pathway_analysis/output/nzero_variants_101.rdata")
DT::datatable(summary_snp_all, options = list(pageLength =20))
```

To compare the p-values of training set and test set, we made a scatter plot.

<figure class="half">
    <img src="https://github.com/xsun1229/pathway_analysis/raw/master/output/LV89_pvals_allpath.jpeg" width="80%">
</figure>


